src/app-layer.c

```diff
@@ -648,6 +678,80 @@ void AppLayerRegisterGlobalCounters(void)
     StatsRegisterGlobalCounter("http.memcap", HTPMemcapGlobalCounter);
 }
 
+void AppLayerSetupCounters(uint8_t ipproto)
+{
+    AppProto alproto;
+    AppProto alprotos[ALPROTO_MAX];
+    uint8_t other_ipproto = (ipproto == IPPROTO_TCP) ? IPPROTO_UDP : IPPROTO_TCP;
+    const char *ipproto_suffix = (ipproto == IPPROTO_TCP) ? "_tcp" : "_udp";
+
+    AppLayerProtoDetectSupportedAppProtocols(alprotos);
+
+    for (alproto = 0; alproto < ALPROTO_MAX; alproto++) {
+        if (alprotos[alproto] == 1) {
+            char *str = "app_layer.flow.";
+            char *alproto_str = AppLayerGetProtoName(alproto);
+            int alproto_len = strlen(alproto_str) + 1;
+            uint8_t ipproto_map = FlowGetProtoMapping(ipproto);
+
+            if (AppLayerParserProtoIsRegistered(ipproto, alproto) &&
```

Comment:
'other proto' is the opposite of ipproto. It's used to check if handle the case that a proto is registered for tcp and udp.